sql/privileges/privileges.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Marc Berhault (marc@cockroachlabs.com)
+
+package privileges
+
+import (
+	"sort"
+	"strings"
+)
+
+// Privilege defines a privilege. This is output by the parser,
+// and used to generate the privilege bitfields in the PrivilegeDescriptor.
+type Privilege uint32
+
+// List of privileges. ALL is specifically encoded so that it will automatically
+// pick up new privileges.
+// TODO(marc): deprecate READ|WRITE.
+const (
+	_             = iota
+	ALL Privilege = iota
```

Comment:
iota is required when a type is specified. see: http://play.golang.org/p/YTj7nE_0Q3

Putting the bit-shifted values in privileges just saves me the shifting when converting to proto, I'd rather not expose it.
Making `ALL = ^0` doesn't help much, I'll still need just as much custom logic just for `ALL`. It would also mean that `ALL` is identifiable only be all bits being set, as opposed to its own value. I'd rather be explicit.